apps/organisations/views.py

```diff
@@ -215,44 +218,93 @@ def get_context_data(self, **kwargs):
 
     def form_valid(self, form):
         data = form.cleaned_data
-        img = self.generate_image(data)
         context = self.get_context_data()
+        img = self.generate_image(data)
+        # get the content form with unchanged cleaned data, as image generation
+        # changes the image data (why?)
+        content_form = CommunicationContentCreationForm(
+            initial=data, project=self.project, format=self.format)
         context['image_preview'] = img
+        context['content_form'] = content_form
         return self.render_to_response(context)
 
     def generate_image(self, data):
-        # Dummy images
-        image = Image.new('RGB', (304, 192), color='red')
-        logo1 = Image.new('RGB', (100, 10), color='blue')
-        logo2 = Image.new('RGB', (100, 10), color='green')
-
-        # Adding padding
+        # Get required image size and image
+        sharepic_format = SOCIAL_MEDIA_SIZES[self.format]
+        image_get = Image.open(data['image'].file)
+        image_size = (sharepic_format['img_min_width'],
+                      sharepic_format['img_min_height'])
+        image = image_get.resize(image_size)
+
+        # Get required total size and add appropriate padding
         right = 0
         left = 0
-        top = 64
-        bottom = 48
+        top = 0
+        bottom = (
+            sharepic_format['overall_height'] -
+            sharepic_format['img_min_height'])
         width, height = image.size
         new_width = width + right + left
         new_height = height + top + bottom
         result = Image.new(
             image.mode, (new_width, new_height), (255, 255, 255))
         result.paste(image, (left, top))
 
-        # Adding writting
-        # Image is converted into editable form using Draw function and
-        # assigned to d1
-        d1 = ImageDraw.Draw(result)
-        # Text location, color and font
+        # image is converted into editable form using Draw function
+        image = ImageDraw.Draw(result)
+
+        # text dimension, color and font
         font = ImageFont.truetype(
-            "adhocracy-plus/assets/fonts/SourceSansPro-Semibold.otf", 20)
+            "adhocracy-plus/assets/fonts/SourceSansPro-Semibold.otf",
```

Comment:
we should use the same way as proposed for the a+ logo to get the path (or use this way for the logo), just for consistency